apps/backend/src/modules/groups/group.service.ts

```diff
@@ -0,0 +1,63 @@
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { ApplicationError } from "~/libs/exceptions/exceptions.js";
+import { transformToSnakeCase } from "~/libs/helpers/helpers.js";
+import { type Service } from "~/libs/types/service.type.js";
+
+import { type GroupCreateRequestDto } from "./libs/types/types.js";
+import { UserGroupModel } from "./user-group.model.js";
+import { UserGroupsToPermissionsModel } from "./user-groups-to-permissions.model.js";
+import { UsersToUserGroupModel } from "./users-to-user-group.model.js";
+
+class GroupService implements Service {
+	public async create(payload: GroupCreateRequestDto): Promise<void> {
+		const { name, permissionIds, userIds } = payload;
+		const key = transformToSnakeCase(name);
+
+		const existingGroup = await UserGroupModel.query().findOne({ key });
```

Comment:
Services shouldn't work with database. You have to create a repository for it

Reply:
That's a good point, but the issue only mentioned adding services and models, so I initially did it this way.